File: src/main/java/com/converter/api/controller/AuthenticationController.java
package com.converter.api.controller;

import com.converter.api.dto.AuthForm;
import com.converter.api.dto.AuthView;
import com.converter.api.service.TokenService;
import org.springframework.http.ResponseEntity;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.AuthenticationException;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

@RestController
@RequestMapping("/auth")
public class AuthenticationController {

    private final AuthenticationManager authManager;
    private final TokenService tokenService;

    public AuthenticationController(AuthenticationManager authManager, TokenService tokenService) {
        this.authManager = authManager;
        this.tokenService = tokenService;
    }

    @PostMapping
    public ResponseEntity<AuthView> autenticar(@Valid @RequestBody AuthForm form) {
        UsernamePasswordAuthenticationToken dadosLogin = form.toUsernamePasswordAuthenticationToken();

        try {
            Authentication authentication = authManager.authenticate(dadosLogin);
            String token = tokenService.generateToken(authentication);
            return ResponseEntity.ok(new AuthView(token, "Bearer"));
        } catch (AuthenticationException e) {
            return ResponseEntity.badRequest().build();
        }
    }

}

File: src/main/kotlin/com/converter/api/controller/AuthenticationController.kt
package com.converter.api.controller

import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import javax.validation.Valid
import com.converter.api.dto.AuthForm
import org.springframework.http.ResponseEntity
import com.converter.api.dto.AuthView
import com.converter.api.service.TokenService

@RestController
@RequestMapping("/auth")
class AuthenticationController(
    private val authManager: AuthenticationManager,
    private val tokenService: TokenService
) {

    @PostMapping
    fun authenticate(@RequestBody @Valid form: AuthForm? = null): ResponseEntity<AuthView> {
        val dadosLogin = form?.toUsernamePasswordAuthenticationToken()

        val authentication = authManager.authenticate(dadosLogin)
        val token = tokenService.generateToken(authentication)
        return ResponseEntity.ok(AuthView(token, "Bearer"))
    }
}